examples/nlp/text_generation_gpt.py

```diff
@@ -0,0 +1,418 @@
+"""
+Title: Simple GPT Text Generation with KerasNLP transformers
+Author: [Jesse Chan](https://github.com/jessechancy)
+Date created: 2022/07/07
+Last modified: 2022/07/07
+Description: Use KerasNLP transformers to train a mini-GPT model for text generation.
+"""
+
+"""
+## Introduction
+
+In this example, we will use KerasNLP layers to build a scaled down Generative
+Pre-trained (GPT) model. GPT is a transformer based model that allows you to generate
+sophisticated text from a small input.
+
+We will train the model on the [simplebooks-92](https://arxiv.org/abs/1911.12391) corpus,
+which is a dataset made from several novels. It is a good dataset for this example since
+it has a small vocabulary and high word frequency, which is beneficial when training a
+model with few parameters.
+
+This example combines concepts from [Text generation with a miniature GPT](https://keras.io/examples/generative/text_generation_with_miniature_gpt/) 
+with KerasNLP abstractions. We will demonstrate how KerasNLP tokenization, model, metrics, and
+inference libraries can be used to simplify the development process. 
+
+The main KerasNLP features shown in this example are:
+
+- [WordPieceTokenizer](https://keras.io/api/keras_nlp/tokenizers/word_piece_tokenizer/):
+An efficient sub-word tokenizer used by BERT and other models.
+- [TokenAndPositionEmbedding](https://keras.io/api/keras_nlp/layers/token_and_position_embedding/): 
+A layer that learns an embedding based on the token and it's position in the
+input.
+- [TransformerDecoder](https://keras.io/api/keras_nlp/layers/transformer_decoder/): A
+decoder transformer layer with causal masking, as described in the transformers paper
+[Attention Is All You Need](https://arxiv.org/abs/1706.03762?context=cs).
+- [Text Generation Utilities](https://github.com/keras-team/keras-nlp/blob/v0.3.0/keras_nlp/utils/text_generation.py): 
+Utilities for different algorithms for text generation, including Beam Search,
+Top-K and Top-P.
+"""
+
+"""
+# Mini-GPT Text Generation with KerasNLP transformers
+
+This example requires KerasNLP. You can install it via the following command: `pip
+install keras-nlp`
+"""
+
+"""
+## Imports
+"""
+
+import os
+import keras_nlp
+import tensorflow as tf
+from tensorflow import keras
+
+"""
+## Settings / Hyperparameters
+"""
+
+# Data
+SEQ_LEN = 128
+MIN_TRAINING_SEQ_LEN = 450
+
+# Inference
```

Comment:
maybe reorder the Hyperparameters blocks to

data
model
training
inference

Reply:
edited